2024/calcite-react/demo/src/Components/View/View.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useRef, useState } from "react"
+
+import WebMap from "@arcgis/core/WebMap";
+import MapView from "@arcgis/core/views/MapView";
+
+import "./View.scss";
+
+interface ViewProps {
+    webmap: string;
+}
+
+export const View = (props: ViewProps) => {
+    const viewRef = useRef(null) as React.RefObject<HTMLDivElement>;
+    const [ view, setView ] = useState<__esri.MapView | null>(null);
+
+    useEffect(() => {
+        return () => {
+            view?.destroy();
+        }
+    }, [])
+    
+    useEffect(() => {
+        if (!viewRef?.current) return;
+
+        const map = new WebMap({
+            portalItem: {
+                id: props.webmap
+            }
+        });
+
+        const view = new MapView({
+            container: viewRef?.current as HTMLDivElement,
+            map
+        });
+
+        setView(view);
+
+    }, [viewRef]);
```

Comment:
a ref can't be used in the dependency array in React, as ref is always pointing to the same object `{ current: ... }`

if you need to use it in the dependency array, you should use the react state instead (`const [viewContainer,setViewContainer] = React.useState<HTMLDivElement | null>(null);` and `<div ref={setViewContainer} id="viewDiv" />`)

but that is only necessary if the ref element is rendered dynamically - which it isn't in this case

thus the viewRef is going to be set by the time the `useEffect` is called, and so you don't need to provide anything in the dependency array:

```suggestion
    }, []);
```